### PR TITLE
[IMP] {website_}mass_mailing: display social media links in mass mail…

### PR DESCRIPTION
**Pull request by sga-odoo**
Task: https://www.odoo.com/web?debug#id=2036094&action=327&model=project.task&view_type=form&menu_id=4720
PAD: https://pad.odoo.com/p/r.055cc5b21b6129a1efb60d8dcd2f51b2

### CODE REVIEW

File: addons/website_mass_mailing/views/snippets_templates.xml [hunk 1]
```diff
@@ -81,79 +81,35 @@
 </template>
 
 <!-- Extend default mass_mailing snippets with website feature -->
-<template id="social_links">
+<template id="mass_mailing.social_links">
     <t t-if="not website">
         <t t-set="website" t-value="request.env['website'].search([], limit=1)"/>
```
Comment (qsm-odoo) [inline addons/website_mass_mailing/views/snippets_templates.xml]: @Gorash Why did you add this here ? Any reason ?

File: addons/website_mass_mailing/views/snippets_templates.xml [hunk 2]
```diff
@@ -81,79 +81,35 @@
 </template>
 
 <!-- Extend default mass_mailing snippets with website feature -->
-<template id="social_links">
+<template id="mass_mailing.social_links">
```
Comment (qsm-odoo) [inline addons/website_mass_mailing/views/snippets_templates.xml]: Should not replace the original template but extend it with xpaths.

Comment (sga-odoo) [reply addons/website_mass_mailing/views/snippets_templates.xml]: Done changes...!

File: addons/website_mass_mailing/views/snippets_templates.xml [hunk 3]
```diff
@@ -81,78 +81,37 @@
...
-    <xpath expr="//table//td" position="after">
-        <td width="30%" class="text-right o_mail_no_resize">
-            <div class="o_mail_header_social">
-                <t t-call="website_mass_mailing.social_links"/>
-            </div>
-        </td>
-    </xpath>
-</template>
+<template id="website_social_links" name="Socila Link Template" inherit_id="mass_mailing.social_links">
+    <xpath expr="//div[@id='social_links']" position="replace">
```
Comment (qsm-odoo) [inline addons/website_mass_mailing/views/snippets_templates.xml]: Noo :wink: xpath each individual link condition to add the new condition, the point of making this an inheriting view is to not duplicate the link definition in both apps.

**Comment by sga-odoo** [comment]
> You are still duplicating everything... :)
> 
> In mass_mailing, do:
> 
> ```
> <div id="social_links">
>     <t t-set="social_facebook" t-value="company_id.social_facebook"/>
>     <a t-if="social_facebook" t-att-href="social_facebook" ...>...</a>
>     ...
> </div>
> ```
> 
> then in website_mass_mailing, you can properly xpath, duplicating nothing:
> 
> ```
> <xpath expr="//t[@t-set='social_facebook']" position="after">
>     <t t-set="social_facebook" t-value="website.social_facebook or social_facebook"/>
> </xpath>
> ```

changes done according to your suggestion. Thanks!

File: addons/website_mass_mailing/views/snippets_templates.xml (hunk 1 above)
Comment (Gorash) [reply addons/website_mass_mailing/views/snippets_templates.xml]: Remember for mass mailing, but it may be a remnant of refactoring and revert.

File: addons/website_mass_mailing/models/res_company.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+    def _get_social_media_links(self):
+        self.ensure_one()
+        website_id = self.env['website'].get_current_website()
+        return {
+            'social_facebook': website_id.social_facebook or self.social_facebook,
+            'social_linkedin': website_id.social_linkedin or self.social_linkedin,
+            'social_twitter': website_id.social_twitter or self.social_twitter,
+            'social_instagram': website_id.social_instagram or self.social_instagram
+        }
```
Comment (dbeguin) [inline addons/website_mass_mailing/models/res_company.py]: You should call super, then check website_id and update values with website_id.social_* if any.
So we do not mess up with hypothetical future overrides

File: addons/website_mass_mailing/models/res_company.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    def _get_social_media_links(self):
+        social_media_links = super()._get_social_media_links()
+        website_id = self.env['website'].get_current_website()
+        social_media_links.update({
+            'social_facebook': website_id.social_facebook or self.social_facebook,
```
Comment (JKE-be) [inline addons/website_mass_mailing/models/res_company.py:14]: fallback should use the value from super() and not re-do the same.

If a third module change it in an override, you will loose the change.

**Comment by tde-banana-odoo** [comment]
Done with https://github.com/odoo/odoo/pull/51025
